logserver/src/main/java/com/yahoo/logserver/handlers/archive/FilesArchived.java

```diff
@@ -37,19 +37,36 @@
     private long maxAgeDays = 30; // GDPR rules: max 30 days
     private long sizeLimit = 30L * (1L << 30); // 30 GB
 
+    private void run() {
+        try {
+            Thread.sleep(125000); // 2 m 5 s
+            while (true) {
+                while (maintenance()) {
+                    Thread.sleep(2000); // 2 s
+                }
+                Thread.sleep(299000); // approx 5 min
+            }
+        } catch (InterruptedException e) {
+            // just exit thread on interrupt
+        }
+    }
+
     /**
-     * Creates an FilesArchive managing the given directory
+     * Creates an instance of FilesArchive managing the given directory
      */
     public FilesArchived(File rootDir) {
         this.root = rootDir;
-        maintenance();
+        rescan();
+        Thread thread = new Thread(this::run);
```

Comment:
Consider assigning a name to the thread (for easier investigating of stack dumps).